### PR TITLE
Numeric type supported for PostgreSQL

### LINKED ISSUE #15338: BigDecimal type not supported by PostgreSQL
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

JDBC performs an insert operation and does not support the BigDecimal type.

```java
"INSERT INTO t_order (type_decimal) VALUES (?);
preparedStatement.setObject(each.getIndex(), new BigDecimal("11.00"););
```
execution succeed

### Actual behavior

java.lang.NumberFormatException

image

### PR DESCRIPTION
**Pull request by soulasuna**
## Numeric type supported for PostgreSQL

Fix #15338.

Changes proposed in this pull request:
-  Numeric type supported for PostgreSQL.
-  Numeric type supported for PostgreSQL in integration test.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/resources/conf/config-shadow.yaml [hunk 1]
```diff
@@ -173,3 +173,72 @@
 #      type: SIMPLE_HINT
 #      props:
 #        foo: bar
+
```
Comment (TeslaCN) [inline shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/resources/conf/config-shadow.yaml:176]: Is this your local changes?

Comment (soulasuna) [reply shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/resources/conf/config-shadow.yaml:176]: I will fix them.

File: shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class PostgreSQLByteConverterTest {
+    
+    @Test
+    public void assertNumeric() {
+        Collection<BigDecimal> bigDecimals = initBigDecimals();
+        bigDecimals.forEach(each -> assertThat(each.equals(PostgreSQLByteConverter.numeric(PostgreSQLByteConverter.numeric(each))), is(true)));
```
Comment (TeslaCN) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java:34]: Consider using `assertTrue` here.

File: shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.LinkedList;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class PostgreSQLByteConverterTest {
```
Comment (TeslaCN) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java:29]: Consider using `@Parameterized`

File: shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java (hunk 2 above)
Comment (soulasuna) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java]: Fixed

File: shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java (hunk 3 above)
Comment (soulasuna) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java:33]: Fixed

File: shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                new Object[]{new BigDecimal("0.0001"), PostgreSQLByteConverter.numeric(new BigDecimal("0.0001"))},
+                new Object[]{new BigDecimal("9999"), PostgreSQLByteConverter.numeric(new BigDecimal("9999"))},
+                new Object[]{new BigDecimal("9999.0"), PostgreSQLByteConverter.numeric(new BigDecimal("9999.0"))},
+                new Object[]{new BigDecimal("9999.9999"), PostgreSQLByteConverter.numeric(new BigDecimal("9999.9999"))}
+        );
+    }
+    
+    @Test
+    public void assertNumeric() {
+        assertThat(PostgreSQLByteConverter.numeric(input), is(expected));
```
Comment (TeslaCN) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java:53]: The tests cannot prove the result is right. Consider asserting raw `byte[]`.

Comment (soulasuna) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLByteConverterTest.java:53]: Fixed

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**